### DISCUSSION TITLE
Chat history log

### DISCUSSION THREAD

**Discussion by ecker00** [oobabooga/text-generation-webui#3423] [Ideas]
Maybe there exist and extension for this, but I've not seen it. But I'm looking for a chat history like OpenAI's chat GPT interface where there is a sidebar you can see and continue old conversations, as well as creating new conversations. So it's easy to have multiple conversations at the same time.

**Comment by viperwasp** [comment]
That would be interesting. I have a question does Ooba allow you to load in previous conversations? I think it can but does it properly load context etc? If Ooba can load previous conversations than I imagine an extension like how you describe will be simple to code for any actual coder. I can't do it. But it's a neat idea.

**Comment by oobabooga** [comment]
You can save and load the entire session (including the chat history) in the Session tab. Make sure to `git pull` or update your installation because this feature has been improved yesterday.

**Reply by viperwasp** [reply]
Thanks I use it often but only to save and look at chats. Thanks for confirming it can load. And also that it's been improved. I've updated. Ty again Oobabooga!

**Comment by rhanna1001** [comment]
Building on this request - 

We run textgen locally at work for a small team of users. Right now, chats do not persist when I leave a session or start a new conversation. 
Is it possible to add a feature so each user can:

1. Access and review their own chat history across sessions
2. Create new conversations without losing previous ones
3. Have their chats stored separately from other users

Is there any roadmap or consideration for adding per-user chat retention?

**Comment by myytautowork-ux** [comment]
I've been using the anilab app for a while now, and one feature that really stood out to me was its ability to connect people [anilab apk today](https://anilabapps.com/) across different communities and platforms. The way it facilitates this connection is through its robust user interface and seamless integration with other apps. How do you think this has impacted the way people interact with each other online?
